src/misc/setup.cpp

```diff
@@ -1573,6 +1573,55 @@ bool Config::ParseConfigFile(const std::string& type,
 	return true;
 }
 
+// Applies queued conf values (i.e.: those that can't be set at runtime but have
+// been queued for config writing or applied on restart), to the given CLI
+// arguments.
+//
+// The function also removes prior duplicates to avoid unecessary redundancy.
+// For example, if a user repeatedly sets the machine type followed by CONFIG
+// -r, then we only want the latest "--set dosbox machine=value" on the CLI
+// and not prior values. So this scans for strips any prior matches.
+//
+void Config::ApplyQueuedValuesToCli(std::vector<std::string>& args) const
+{
+	for (const auto section : sectionlist) {
+		const auto properties = dynamic_cast<Section_prop*>(section);
+		if (!properties) {
+			continue;
+		}
+
+		for (const auto property : *properties) {
+			const auto queued_value = property->GetQueuedValue();
+			if (!queued_value) {
+				continue;
+			}
+
+			// Format the section and property portion
+			auto sec_prop = format_str("%s %s=",
+			                           section->GetName(),
+			                           property->propname.c_str());
+
+			// Remove any existing matches
+			auto it = args.begin();
+			while (it != args.end() && std::next(it) != args.end()) {
+				const auto current = it;
+				const auto next    = std::next(it);
+
+				if (*current == "--set" &&
+				    next->starts_with(sec_prop)) {
+					it = args.erase(current, std::next(next));
+				} else {
+					++it;
+				}
+			}
+
+			// Add the new arguments with the queued value
+			args.emplace_back("--set");
+			args.emplace_back(sec_prop + *queued_value);
+		}
+	}
+}
```

Comment:
Right... I think it would be better to have each config setting have a "slot" for its "future queued value" instead of having this separate vector. I'm just thinking of the GUI; I'd like to be able to call `HasQueuedSetting()` at a per-property level.

But many changes might be necessary for the GUI stuff, so this is fine for now. Even if you reimplemented it like I explained, I might need to thoroughly refactor the entire config subsystem... So probably this is fine until then.

Reply:
Yup, this PR adds this slot; the config property class now has a std::optional queueable_value private member, with Set and Get function.

This snippet you've flagged here modifies the to-be-used CLI arguments with the queued values. (And the CLI args are just a string.. so that's why it's appending). 

The restart action sets up the new process + argv, launches the process, and then the current dosbox process dies.

 So the only way to convey information (the new conf settings) to the new PID is via argv (or some temp file, which is much uglier 😅 )

Reply:
Right, I kinda missed that part 😄

Reply:
Ok... It has finally dawned on me why this thing is needed. We don't restart "internally", but we literally quit and restart the process. Got it 😄